lectures/supply_demand_multiple_goods.md

```diff
@@ -465,16 +902,190 @@ We'll soon see that a monopolist sets a **lower output** $q$ than does either a
 
 
 ```{exercise}
-:label: sdm_ex3
+:label: sdm_ex4
 
 Please  verify the monopolist's supply curve {eq}`eq:qmonop`.
 
 ```
 
++++
 
+### A monopolist
 
+Let's consider a monopolist supplier.
+
+We have included a method in our `ProductionEconomy` class to compute an equilibrium price and allocation when the supplier is a monopolist.
+
+Since the supplier now has the price-setting power
+
+- we first compute the optimal quantity that solves the monopolist's profit maximization problem.
+- Then we back out an equilibrium price from the consumer's inverse demand curve.
+
+Next, we use a graph for the single good case to illustrate the difference between a competitive equilibrium and an equilibrium with a monopolist supplier.
+
+Recall that in a competitive equilibrium, a price-taking supplier equates marginal revenue $p$ to marginal cost $h + Hq$.
+
+This yields a competitive producer's inverse supply curve.
+
+A monopolist's marginal revenue is not constant but instead is a non-trivial function of the quantity it sets.
+
+The monopolist's marginal revenue is
+
+$$
+MR(q) = -2\mu^{-1}\Pi^{\top}\Pi q+\mu^{-1}\Pi^{\top}b,
+$$
+
+which the monopolist equates to its marginal cost.
+
+The plot indicates that the monopolist's sets output  lower than either the competitive equilibrium quantity.
+
+In a single good case, this equilibrium is associated with a higher price of the good.
+
+```{code-cell} ipython3
+class Monopoly(ProductionEconomy):
```

Comment:
Thanks for your advice today @jstac . 

Here I define an inheritance class called ``Monopoly``.